tests/unit/delegate.ts

```diff
@@ -0,0 +1,180 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import { emit } from 'dojo-core/on';
+import { Handle } from 'dojo-core/interfaces';
+import delegate from 'src/delegate';
+import * as dom from 'src/dom';
+
+let container = document.createElement('div');
+let handles: (Handle)[] = [];
+
+function testDelegate(...args: any[]) {
+	let handle = delegate.apply(null, arguments);
+	handles.push(handle);
+	return handle;
+}
+
+registerSuite({
+	name: 'delegate',
+
+	setup() {
+		document.body.appendChild(container);
+	},
+
+	teardown() {
+		document.body.removeChild(container);
+	},
+
+	afterEach() {
+		while (handles.length > 0) {
```

Comment:
Nit-picking, but this can use for...of to avoid manipulating the array every time (then just reset handles to a new array).  (I'm fixing this on my branch.)